Kickstarter-iOS/Features/PledgeView/Controllers/PostCampaignCheckoutViewController.swift

```diff
@@ -402,38 +403,83 @@ extension PostCampaignCheckoutViewController: PledgeViewControllerMessageDisplay
   }
 }
 
-// MARK: - MessageBannerViewControllerDelegate
+// MARK: - STPApplePayContextDelegate
 
-extension PostCampaignCheckoutViewController: PKPaymentAuthorizationViewControllerDelegate {
-  func paymentAuthorizationViewControllerDidFinish(_ controller: PKPaymentAuthorizationViewController) {
-    controller.dismiss(animated: true, completion: { [weak self] in
-      self?.viewModel.inputs.paymentAuthorizationViewControllerDidFinish()
-    })
-  }
+enum PostCampaignCheckoutApplePayError: Error {
+  case missingToken(String)
+  case missingPaymentMethodInfo(String)
+  case missingPaymentIntent(String)
+}
 
-  func paymentAuthorizationViewController(
-    _: PKPaymentAuthorizationViewController,
-    didAuthorizePayment payment: PKPayment,
-    handler completion: @escaping (PKPaymentAuthorizationResult)
-      -> Void
+extension PostCampaignCheckoutViewController: STPApplePayContextDelegate {
+  func applePayContext(
+    _: StripeApplePay.STPApplePayContext,
+    didCreatePaymentMethod _: StripePayments.STPPaymentMethod,
+    paymentInformation payment: PKPayment,
+    completion: @escaping StripeApplePay.STPIntentClientSecretCompletionBlock
   ) {
-    let paymentDisplayName = payment.token.paymentMethod.displayName
-    let paymentNetworkName = payment.token.paymentMethod.network?.rawValue
+    guard let paymentIntentClientSecret = self.applePayPaymentIntent else {
+      completion(
+        nil,
+        PostCampaignCheckoutApplePayError.missingPaymentIntent("Missing PaymentIntent")
+      )
+      return
+    }
```

Comment:
Stripe assumes you'll request the payment intent asynchronously from inside this method, but in our case, we request it immediately when you tap the ApplePay button.